### DISCUSSION TITLE
is that possible to map @db.varchar(length) to typescript

### DISCUSSION THREAD

**Discussion by kin-for-test** [prisma/prisma#18896] [Q&A] [answered]
```prisma
model User {
  name                String                     @db.VarChar(30)
}
```

to typescript like :

```js
import type { User } from "@prisma/client";
console.log(User.nameMaxLength)  // 30
```

**Comment by ludralph** [comment, marked as answer]
Hi @kin-for-test 👋 

It's not currently possible to access the length of a @db.VarChar field in TypeScript using the Prisma Client's generated types. It currently does not generate a type for the maximum length of a VarChar field. Instead, you can use the length property of the string when setting its value in your application code. for example
```js
 await prisma.user.create({
    data: {
      name: name.substring(0, 30), // truncate name to 30 characters
    },
  });
```

**Reply by kin-for-test** [reply]
thank! @ludralph 

this may be a work around , but this is not type safe .
is there a plan to add this feature in the future ?

**Reply by ludralph** [reply]
Hi @kin-for-test, You can open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) here documenting your usecase so our engineers can take a look.

**Reply by ludralph** [reply]
In addition, you might want to check out Prisma client [extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions) and perform the validation there. Here is an [example](https://github.com/prisma/prisma-client-extensions/tree/main/input-validation)
